__tests__/10_deep_proxy_spec.ts

```diff
@@ -162,6 +162,17 @@ describe('keys spec', () => {
     expect(isChanged(s1, { a: s1.a }, a1)).toBe(false);
     expect(isChanged(s1, { c: 'c', d: 'd' }, a1)).toBe(true);
   });
+
+  it('hasOwnProperty', () => {
+    const s1 = { a: { b: 'b' }, c: 'c' };
+    const a1 = new WeakMap();
+    const p1 = createProxy(s1, a1);
+    // eslint-disable-next-line no-prototype-builtins
+    noop(p1.hasOwnProperty('a'));
+    expect(isChanged(s1, { a: s1.a, c: 'c' }, a1)).toBe(false);
+    expect(isChanged(s1, { a: s1.a }, a1)).toBe(false);
+    expect(isChanged(s1, { c: 'c', d: 'd' }, a1)).toBe(true);
+  });
```

Comment:
Might consider adding a case for the limitation highlighted in `getOwnPropertyDescriptor` e.g.

```suggestion
    // NOTE: due to the way this is implemented, changing the value of 'a' causes 
    // a change, even thought we only used `hasOwnProperty` above
    expect(isChanged(s1, { a: null, c: 'c' }, a1)).toBe(true);
  });
```

Reply:
Yeah, we could add it. Although it's not an ideal spec, it's the current LIMITED spec. So, why not..